src/community/csw-iso/src/main/java/org/geoserver/csw/response/MetaDataTransformer.java

```diff
@@ -74,6 +76,20 @@ private void encodeProperty(Feature f, Property p) {
                         MetaDataDescriptor.NAMESPACES.getPrefix(p.getName().getNamespaceURI());
 
                 AttributesImpl atts = new AttributesImpl();
+                if (p.isNillable()) {
+                    Property prop =
+                            ((ComplexAttribute) p)
+                                    .getProperty(ComplexFeatureConstants.SIMPLE_CONTENT);
+                    boolean nil =
+                            prop == null || prop.getValue() == null || prop.getValue().equals("");
```

Comment:
You could use here the same ``Strings.isNullOrEmpty`` you used bellow.

Reply:
no bcause getValue() is not necessarily a String... so we have to check if it is null (of any type) or equal to the empty string